spec/javascript/components/molecules/StatusDropdown.test.js

```diff
@@ -0,0 +1,130 @@
+      expect(toggleButton.text()).toEqual('Status')
+      expect(toggleButton.hasClass(expectedButtonClasses.join(' '))).toEqual(true)
+      expect(toggleButton.prop('disabled')).toEqual(false)
+    })
+
+    test('without any additional button styles', () => {
+      wrapper = getWrapper()
+      const toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.hasClass('expand')).toBe(false)
+      expect(toggleButton.hasClass('tiny')).toBe(false)
+      expect(toggleButton.hasClass('small')).toBe(false)
+    })
+
+    test('when additional styles are provided', () => {
+      wrapper = getWrapper({ expand: true, size: 'tiny' })
+      let toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.hasClass('expand')).toBe(true)
+      expect(toggleButton.hasClass('tiny')).toBe(true)
+      expect(toggleButton.hasClass('small')).toBe(false)
+
+      wrapper = getWrapper({ expand: true, size: 'small' })
+      toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.hasClass('tiny')).toBe(false)
+      expect(toggleButton.hasClass('small')).toBe(true)
+    })
+
+    test('when value provided and is styled', () => {
+      wrapper = getWrapper({ status: 'Appealed' })
+      const expectedButtonClasses = [
+        'button',
+        'dropdown-button',
+        'has-icon--right',
+        'text-align-left',
+        'is-appealed'
+      ]
+      const toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.text()).toEqual('Appealed')
+      expect(toggleButton.hasClass(expectedButtonClasses.join(' '))).toBe(true)
+    })
+
+    test('when the dropdown is disabled', () => {
+      wrapper = getWrapper({ disabled: true })
+      const toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.prop('disabled')).toEqual(true)
+    })
+
+    test('when a placeholder is provided', () => {
+      wrapper = getWrapper({ placeholder: 'placeholder' })
+      const toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.text()).toEqual('placeholder')
+    })
+  })
+
+  describe('it renders the options as expected', () => {
+    const getStatusOptionWrapper = (option, selectedValue) => shallow(
+      renderStatusOption(option, { selectValue: [{ value: selectedValue }] })
+    )
+    const testValue = 'testValue'
+    const testStatusClassName = 'status-class-name'
+    const testLabel = 'testLabel'
+    const sampleOption = {
+      value: testValue,
+      label: testLabel,
+      statusClassName: testStatusClassName
+    }
+
+    test('with a styled value', () => {
+      wrapper = getStatusOptionWrapper(sampleOption, 'notSelected')
+      const expectedLiClasses = `dropdown-menu_item ${testStatusClassName}`
```

Comment:
Another case where I'm concerned that we're checking ordering where we don't really care what order the classes are in. Could this be two separate expects, one for each class we want to check?